packages/itwinui-react/src/core/Table/TablePaginator.tsx

```diff
@@ -252,7 +254,12 @@ export const TablePaginator = (props: TablePaginatorProps) => {
         )}
       </Box>
       {showPagesList && (
-        <OverflowContainer className='iui-center' itemsCount={totalPagesCount}>
+        <OverflowContainer
+          key={centerContainerRef.current?.scrollWidth}
```

Comment:
ref is being read during render.

Can we handle this some other way? Would `paginatorWidth` work as a key?

Reply:
Maybe you can replace `useContainerWidth` with a `useResizeObserver` that lets you save both the full width and the `scrollWidth` into state?

Reply:
I'm also suggesting removing `useContainerWidth` (which itself uses `useResizeObserver`). We shouldn't set up two resize observers, we can use `useResizeObserver` to save all the states that we need.

Reply:
So then this approach will not work. I'd say let's revisit this after #2154.

For now, can we use `z-index` as a workaround? The center portion can be given a higher `z-index` than the right portion, to ensure the buttons are interactable even when they overlap. It's not perfect, but better than current situation.